Source/Fuse.Reactive.JavaScript/Observable.uno

```diff
@@ -85,19 +119,31 @@ namespace Fuse.Reactive
 					Fuse.Diagnostics.InternalError( "Unexpected null object", this );
 					return;
 				}
-				
-				try
+
+				var op = new SetExclusiveOperation(_om._worker, _om.Object, newValue, _origin);
+				if (_om._worker.CanEvaluate)
```

Comment:
I like this pattern, but it seems to be repeated in a few places - any way we can get some reuse/abstraction here perhaps? Seems like this would be easy to break in the future.

Reply:
It's not quite clear to me what you mean. If you think there's too much repetition here in checking CanEvaluate before dispatching, I kinda agree, but there's a kicker: According to @sebbert, we don't really need all three, as ReplaceAllExclusive() can accurately express all three cases anyway. But changing that is a separate change that I'd rather not have to depend on in order to restore thread-sanity. So while I agree, I'm not sure it's worth the effort, because I expect a lot of this will go away.

What I really dislike here, is not having any safety-net to catch cases like this. I kind of want to refactor things a bit to make it API-wise impossible to call JS from the UI thread, but again that's a bigger change.

Reply:
That is indeed what I meant, that it's repetitive and I'm worried that it will be easy to break one of them in the future. But I think dealing with that in a separate change makes perfect sense 👍

Reply:
I'm a bit surprised that this is ever called from the JavaScript thread. These are meant as Uno APIs to update JS. They could probably just always dispatch to the other thread, since I don't know that they have another appropriate use-case.